src/test/js-transform_test.ts

```diff
@@ -124,92 +124,119 @@ suite('jsTransform', () => {
+      `);
+
+      const expected = stripIndent(`
+        import { dep1 } from './node_modules/dep1/index.js';
+        import { dep2 } from './node_modules/dep2/dep2.js';
+        import { dep2A } from './node_modules/dep2/a.js';
+        import { dep3 } from './node_modules/dep3/dep3-module.js';
+        import { dep4 } from './node_modules/dep4/dep4-module.js';
+      `);
+
+      const result = jsTransform(input, {moduleResolution: 'node', filePath});
+      assert.equal(result.trim(), expected.trim());
+    });
 
-    const expected = stripIndent(`
-      import { dep1 } from './node_modules/dep1/index.js';
-      import { dep2 } from './node_modules/dep2/dep2.js';
-      import { dep2A } from './node_modules/dep2/a.js';
-      import { dep3 } from './node_modules/dep3/dep3-module.js';
-      import { dep4 } from './node_modules/dep4/dep4-module.js';
-
-      import { p1 } from '/already/a/path.js';
-      import { p2 } from './already/a/path.js';
-      import { p3 } from '../already/a/path.js';
-      import { p4 } from '../already/a/path.js';
-      import { p5 } from 'http://example.com/already/a/path.js';
-    `);
+    test('regular paths and urls', () => {
+      const filePath = path.join(fixtureRoot, 'foo.js') as LocalFsPath;
+
+      const input = stripIndent(`
+        import { p1 } from '/already/a/path.js';
+        import { p2 } from './already/a/path.js';
+        import { p3 } from '../already/a/path.js';
+        import { p4 } from '../already/a/path.js';
+        import { p5 } from 'http://example.com/already/a/path.js';
+      `);
+
+      const expected = stripIndent(`
+        import { p1 } from '/already/a/path.js';
+        import { p2 } from './already/a/path.js';
+        import { p3 } from '../already/a/path.js';
+        import { p4 } from '../already/a/path.js';
+        import { p5 } from 'http://example.com/already/a/path.js';
+      `);
+
+      const result = jsTransform(input, {moduleResolution: 'node', filePath});
+      assert.equal(result.trim(), expected.trim());
+    });
 
-    const result = jsTransform(input, {moduleResolution: 'node', filePath});
-    assert.equal(result.trim(), expected.trim());
-  });
+    test('paths that still need node resolution', () => {
+      const filePath = path.join(fixtureRoot, 'foo.js') as LocalFsPath;
 
-  test('rewrites bare module specifiers to paths for dependencies', () => {
-    const fixtureRoot =
-        path.join(__dirname, '..', '..', 'test-fixtures', 'npm-modules');
-    const filePath = path.join(fixtureRoot, 'npm-module.js') as LocalFsPath;
+      const input = stripIndent(`
+        import { bar } from './bar';
+        import { baz } from './baz';
+        import { qux } from './qux';
+      `);
 
-    const input = stripIndent(`
-      import { dep1 } from 'dep1';
-    `);
+      const expected = stripIndent(`
+        import { bar } from './bar.js';
+        import { baz } from './baz.json';
+        import { qux } from './qux';
```

Comment:
It's testing that if the exact extension-less file exists, it will be preferred over .js (which is what Node does). I split it into a separate test to self-document it.